### PR TITLE
feature-MOVE-1222: Added rollback functionality

### PR DESCRIPTION
**Pull request by FrodeBjerkholt**
Added blacklisting when upgrade fails.
 Added mail reports when upgrading/rollback.
 Added better communication with the actuator-endpoint.
 Added scanning of the startup-log to detect success/failure
 Some refactoring.
 Some Sonar pleasing.

### CODE REVIEW

**Comment by FrodeBjerkholt** [comment]
Det er mange endringer, så det enkleste er nok å sjekke ut branchen i stedet for å se på GitHub.

File: deploymanager/src/main/java/no/difi/move/deploymanager/service/laucher/LauncherServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+            switch (waitForStartup(startupLog)) {
+                case SUCCESS:
+                    log.info("Application started successfully!");
+                    break;
+                case FAILED:
+                    log.error("Application failed!");
+                    future.cancel(true);
+                    break;
+                case UNKONWN:
```
Comment (johannmo) [inline deploymanager/src/main/java/no/difi/move/deploymanager/service/laucher/LauncherServiceImpl.java]: Ein liten skrivefeil her.

Comment (FrodeBjerkholt) [reply deploymanager/src/main/java/no/difi/move/deploymanager/service/laucher/LauncherServiceImpl.java]: Retter det

File: deploymanager/src/main/java/no/difi/move/deploymanager/service/mail/MailServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Override
+    public void sendMail(String subject, String content) {
+        DeployManagerProperties.MailProperties mail = properties.getMail();
+
+        if (mail == null) {
+            return;
+        }
+
+        Objects.requireNonNull(mail.getRecipient(), "deploymanager.mail.receipient must be set");
```
Comment (johannmo) [inline deploymanager/src/main/java/no/difi/move/deploymanager/service/mail/MailServiceImpl.java]: Liten skrivefeil i receipient og.

File: deploymanager/src/main/java/no/difi/move/deploymanager/repo/DeployDirectoryRepo.java [hunk 3]
```diff
@@ -1,56 +1,84 @@
...
         return propertiesFile;
     }
 
-    private File getRoot() {
+    private File getOrCreateRoot() {
         File root = new File(properties.getRoot());
-        if (!root.exists()) {
-            root.mkdir();
+        if (root.mkdir()) {
+            log.info("Created rppt folder: {}", root.getAbsolutePath());
```
Comment (johannmo) [inline deploymanager/src/main/java/no/difi/move/deploymanager/repo/DeployDirectoryRepo.java]: Skrivefeil i rppt og.

**Pull request merged**